### ISSUE TITLE
UrbanSim using outdated version of Patsy

### ISSUE THREAD

**Issue by apdjustino** [UDST/urbansim#169]
I discovered this error while trying to fit a SegmentedMNLDiscreteChoiceModel. Patsy throws an error when creating the dmatrix. Looking into it I found that the author of Patsy fixed this when upgrading Patsy from v.0.4.0 to v0.4.1 

I upgraded Patsy and this corrected the error. I don't know if the latest UrbanSim release bundles Patsy 0.4.1, but it will need the newer version of the package to fit models.

**Comment by pksohn** [comment]
@apdjustino I know it's been a while, but do you remember if the segmented MNL model was indeed breaking on v0.4.0, or were you maybe on an earlier version? I'm able to reproduce this error on Patsy v0.2.1 which urbansim currently requires so it's clear we need to bump the required version, but for me the issue is resolved upon upgrading to v0.3.0.

**Comment by pksohn** [comment]
We're going to update the Patsy required version to `0.3.0` in this release. Give us a heads up if you still think we need to require a higher version.

**Comment by pksohn** [comment]
Addressed with #195.

**Comment by pksohn** [comment]
Reopening this as @sablanchard found a similar bug. 

We should likely bump required version of Patsy all the way up to `0.4.1` to ensure the [bugfix](https://patsy.readthedocs.io/en/latest/changes.html#v0-4-1) is included, as @apdjustino suggested.

```
There are 697875 total available units
    and 118807 total choosers
    but there are 0 overfull buildings
    for a total of 580244 temporarily empty units
    in 10410 buildings total in the region
There are 1176 total movers for this LCM
Traceback (most recent call last):
  File "simulate.py", line 172, in <module>
    run()
  File "simulate.py", line 94, in run
    orca.run(datasources.step_sequence(), iter_vars = range(2011, forecast_year + 1))
  File "C:\Users\Sam\Anaconda2\lib\site-packages\orca\orca.py", line 1987, in run
    step()
  File "C:\Users\Sam\Anaconda2\lib\site-packages\orca\orca.py", line 792, in __call__
    return self._func(**kwargs)
  File "C:\Users\Sam\Dropbox\Work\github_projects\block_model\block_model\models.py", line 1145, in func
    remaining_capacity_attrib)
  File "C:\Users\Sam\Dropbox\Work\github_projects\block_model\block_model\utils.py", line 352, in lcm_simulate
    new_units, _ = yaml_to_class(cfg).predict_from_cfg(movers, units, cfgname=cfg, alternative_ratio=alternative_ratio)
  File "c:\repos\ual\urbansim\urbansim\models\dcm.py", line 830, in predict_from_cfg
    new_units = lcm.predict(choosers, alternatives, debug=debug)
  File "c:\repos\ual\urbansim\urbansim\models\dcm.py", line 635, in predict
    choosers, alternatives, filter_tables=False)
  File "c:\repos\ual\urbansim\urbansim\models\dcm.py", line 524, in probabilities
    self.str_model_expression, data=merged, return_type='dataframe')
  File "C:\Users\Sam\Anaconda2\lib\site-packages\patsy\highlevel.py", line 278, in dmatrix
    NA_action, return_type)
  File "C:\Users\Sam\Anaconda2\lib\site-packages\patsy\highlevel.py", line 152, in _do_highlevel_design
    NA_action)
  File "C:\Users\Sam\Anaconda2\lib\site-packages\patsy\highlevel.py", line 57, in _try_incr_builders
    NA_action)
  File "C:\Users\Sam\Anaconda2\lib\site-packages\patsy\build.py", line 706, in design_matrix_builders
    categories=None)
  File "C:\Users\Sam\Anaconda2\lib\site-packages\patsy\design_info.py", line 88, in __init__
    raise ValueError("For numerical factors, num_columns "
ValueError: For numerical factors, num_columns must be an int
Closing remaining open files:.\data\model_data.h5...done
```

**Comment by smmaurer** [comment]
Resolved in PR #225.
